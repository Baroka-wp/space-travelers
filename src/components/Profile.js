import { useSelector } from 'react-redux';
import ProfilRocketCard from './pages/ProfilRocketCard';

const Profile = () => {
  const rockets = useSelector((state) => state.rockets);
  const myRockets = rockets.filter((rocket) => rocket.reserved === true);
  return (
    <div>
      <h2>My Rockets</h2>
      <ProfilRocketCard myRockets={myRockets} />
      <h2>My Missions</h2>
    </div>

  );
};

export default Profile;
