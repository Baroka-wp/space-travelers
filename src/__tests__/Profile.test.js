import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import Profile from '../components/Profile';

describe('Display missions element', () => {
  test('Check if Profile Component Renders in DOM properly', () => {
    render(
      <Provider store={store}>
        <Profile />
      </Provider>,
    );
    expect(screen.queryByText(/My Rockets/)).toBeTruthy();
    expect(screen.queryByText(/My Missions/)).toBeTruthy();
  });
  it('renders properly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Profile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
