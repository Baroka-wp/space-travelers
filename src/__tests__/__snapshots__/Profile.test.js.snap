// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display missions element renders properly 1`] = `
<div
  className="container my_profil"
>
  <div
    className="side_card"
  >
    <h2>
      My Rockets
    </h2>
    <table
      className="profil_Table"
    >
      <tbody />
    </table>
  </div>
  <div
    className="side_card"
  >
    <h2>
      My Missions
    </h2>
    <table
      className="profil_Table"
    >
      <tbody />
    </table>
  </div>
</div>
`;
